framework/data/Sort.php

```diff
@@ -243,8 +243,8 @@ public function getAttributeOrders($recalculate = false)
                 $request = Yii::$app->getRequest();
                 $params = $request instanceof Request ? $request->getQueryParams() : [];
             }
-            if (isset($params[$this->sortParam]) && is_scalar($params[$this->sortParam])) {
-                $attributes = explode($this->separator, $params[$this->sortParam]);
+            if (isset($params[$this->sortParam])) {
```

Comment:
This doesn't look like total equivalent condition in before and after change code...